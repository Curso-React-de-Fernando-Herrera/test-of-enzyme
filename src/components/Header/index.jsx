import PropTypes from 'prop-types'
import React from 'react'

const Header = ({ title, subtitle }) => {
  return <header>
    <h1>{title}</h1>
    <h2>{subtitle}</h2>
  </header>
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired
}

export default Header
