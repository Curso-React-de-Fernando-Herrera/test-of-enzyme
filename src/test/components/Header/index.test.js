import Header from 'components/Header'
import { shallow } from 'enzyme'


describe('<Header /> should be', () => {
  test('a component rendered by title and subtitle', () => {
    const initialTitle = 'Bienvenido!'
    const initialSubtitle = 'Al testing con Enzyme'

    const wrapper = shallow(<Header title={initialTitle} subtitle={initialSubtitle} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('a title arrive on html', () => {
    const initialTitle = 'Bienvenido!'
    const initialSubtitle = 'Al testing con Enzyme'
    const wrapper = shallow(<Header title={initialTitle} subtitle={initialSubtitle} />)

    const textOfTitle = wrapper.find('h1').text()
    const incorrectTitle = wrapper.find('h2').text()

    expect(textOfTitle).toBe(initialTitle)
    expect(incorrectTitle).not.toBe(initialTitle)
  })
})