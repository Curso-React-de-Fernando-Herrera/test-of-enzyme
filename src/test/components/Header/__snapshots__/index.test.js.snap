// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should be a component rendered by title and subtitle 1`] = `
<header>
  <h1>
    Bienvenido!
  </h1>
  <h2>
    Al testing con Enzyme
  </h2>
</header>
`;
